src/Fixer/LanguageConstruct/FunctionToConstantFixer.php

```diff
@@ -119,8 +130,10 @@ public function isRisky()
      */
     protected function applyFix(\SplFileInfo $file, Tokens $tokens)
     {
+        $functionAnalyzer = new FunctionsAnalyzer();
+
         for ($index = $tokens->count() - 4; $index > 0; --$index) {
-            $candidate = $this->getReplaceCandidate($tokens, $index);
+            $candidate = $this->getReplaceCandidate($tokens, $functionAnalyzer, $index);
```

Comment:
Is there a need to pass `$functionAnalyzer`? Does it save us some time/resources?

I see in other methods which are in a loop that we create a new instance of `FunctionsAnalyzer` in the loop. Maybe it's worth to fix those then.

Reply:
I wanted to prevent creating a new instance over and over. In other places this might be nice to clean up, maybe move to a private property of the etc., really depends on each case

Reply:
Talking about clean up I was thinking about passing `$tokens` to the constructor in every analyzer, not only one of them.

Reply:
haven't  looked at the impact, but sounds interesting!